influxql/ast.go

```diff
@@ -1029,40 +1073,91 @@ func (s *SelectStatement) validateFields() error {
 	return nil
 }
 
+func (s *SelectStatement) allowMixedAggregates(numAggregates int) error {
```

Comment:
Sort of.  So clearly the name is horrible.  I'll think on a better one.

In theory `TOP`, `BOTTOM`, `MIN`, `MAX`, `FIRST`, and `LAST` can all be called with additional fields/tags in the query, but other aggregates, like `MEAN` can ONLY be called by themselves, or potentially with other aggregates (I'm not sure about this last part though).

Let me think about how to make this better.